File: app/src/main/java/com/zerostress/manager/models/Season.java
package com.zerostress.manager.models;

public class Season {
    private String id;
    private String name;
    private long startDate;
    private long endDate;
    private boolean active;
    private int topRewardCoins;

    public Season() {}

    public Season(String name, long startDate, long endDate) {
        this.name = name;
        this.startDate = startDate;
        this.endDate = endDate;
        this.active = true;
        this.topRewardCoins = 500;
    }

    public String getId() { return id; }
    public String getName() { return name; }
    public long getStartDate() { return startDate; }
    public long getEndDate() { return endDate; }
    public boolean isActive() { return active; }
    public int getTopRewardCoins() { return topRewardCoins; }

    public void setId(String id) { this.id = id; }
    public void setActive(boolean active) { this.active = active; }
}

File: app/src/main/java/com/zerostress/manager/models/Season.kt
package com.zerostress.manager.models

import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
data class Season(
    var id: String = "",
    var name: String = "",
    var description: String? = null,
    var duration: String? = null,
    var startDate: Long = 0L,
    var endDate: Long = 0L,
    var active: Boolean = true,
    var topRewardCoins: Int = 500,
    var createdAt: Long = 0L
) {
    companion object {
        fun create(name: String, startDate: Long, endDate: Long): Season =
            Season(
                name = name,
                startDate = startDate,
                endDate = endDate,
                active = true,
                topRewardCoins = 500
            )
    }
}
